src/main/kotlin/com/autonomousapps/tasks/BuildHealthTask.kt

```diff
@@ -25,11 +28,41 @@ abstract class BuildHealthTask : DefaultTask() {
   @get:InputFile
   abstract val adviceReport: RegularFileProperty
 
+  @get:Input
+  abstract val dependencyRenamingMap: MapProperty<String, String>
+
   @TaskAction fun action() {
-    val buildHealths = adviceReport.fromJsonList<BuildHealth>()
+    val buildHealth = adviceReport.fromJsonList<ComprehensiveAdvice>()
+
+    val mapper = dependencyRenamingMap.orNull
+    val inputFilePath = adviceReport.get().asFile.path
+
+    buildHealth.forEach { projectAdvice ->
+      val consoleReport = ConsoleReport.from(projectAdvice)
+      val advicePrinter = AdvicePrinter(consoleReport, mapper)
 
-    if (buildHealths.any { it.shouldFail }) {
+      val consoleText = advicePrinter.consoleText()
+      if (shouldNotBeSilent()) {
+        logger.quiet(projectHeaderText(projectAdvice.projectPath))
+        logger.quiet(consoleText)
+        if (consoleReport.isNotEmpty()) {
+          logger.quiet("See machine-readable report at $inputFilePath\n")
```

Comment:
This results in duplicated `See machine-readable report at` for each project. I would suggest to move it out of `forEach`

Reply:
Ah, good catch.